communication/src/ping.h

```diff
@@ -44,6 +50,10 @@ class Pinger
 				// timed out, disconnect
 				return PING_TIMEOUT;
 			}
+			else
+			{
+				expecting_ping_ack = false;
```

Comment:
This shouldn't be here and was specifically removed in 0.8.0-rc.3 as a bug, which now makes me naturally very suspicious of all changes that are not in the tests directly :)

Reply:
Yes, it was this change that I wanted us to review. I don't have any context for the bug or how the behaviour is not correct, and what the desired behaviour should be.

Reply:
I see your original comments now.  It's possible the tests need to be updated and not the code.  But definitely some tests need to be added.